cni/pkg/install-cni/pkg/install/install.go

```diff
@@ -0,0 +1,263 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package install
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"syscall"
+
+	"github.com/coreos/etcd/pkg/fileutil"
+	"github.com/fsnotify/fsnotify"
+	"github.com/pkg/errors"
+
+	"istio.io/istio/cni/pkg/install-cni/pkg/config"
+	"istio.io/istio/cni/pkg/install-cni/pkg/constants"
+	"istio.io/istio/cni/pkg/install-cni/pkg/util"
+	"istio.io/pkg/log"
+)
+
+type configFiles struct {
+	kubeconfigFilepath string
+	cniConfigFilepath  string
+}
+
+// Run starts the installation process with given configuration
+func Run(cfg *config.Config) (err error) {
```

Comment:
This can be written as
```
// use struct to store config and state (if it has)
type installer struct {
  cfg *config.Config
}

func (*installer) Install() error {
}

func (*installer) cleanup() error {
}

```